### PR TITLE
Added mean absolute error in linear regression

### PR DESCRIPTION
**Pull request by KushagraMakharia**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: machine_learning/linear_regression.py [hunk 1]
```diff
@@ -87,6 +87,19 @@ def run_linear_regression(data_x, data_y):
     return theta
 
 
+def mean_absolute_error(pred_X, org_X):
```
Comment (CaedenPH) [inline machine_learning/linear_regression.py:90]: Please provide more descriptive variable names and use `snake_case`

File: machine_learning/linear_regression.py [hunk 2]
```diff
@@ -87,6 +87,19 @@ def run_linear_regression(data_x, data_y):
...
+    """Return sum of square error for error calculation
+    :param predicted_y   : contains the output of prediction (result vector)
+    :param original_y    : contains values of expected outcome
+    :return          : mean absolute error computed from given feature's
+    """
+    total = 0
+    for i in range(len(original_y)):
+        total += abs(original_y[i] - predicted_y[i])
+    error = total / len(original_y)
+    return error
```
Comment (CaedenPH) [inline machine_learning/linear_regression.py:100]:
```suggestion
    return total / len(original_y)
```

File: machine_learning/linear_regression.py [hunk 3]
```diff
@@ -87,6 +87,19 @@ def run_linear_regression(data_x, data_y):
...
 
+def mean_absolute_error(predicted_y, original_y):
+    """Return sum of square error for error calculation
+    :param predicted_y   : contains the output of prediction (result vector)
+    :param original_y    : contains values of expected outcome
+    :return          : mean absolute error computed from given feature's
+    """
+    total = 0
+    for i in range(len(original_y)):
+        total += abs(original_y[i] - predicted_y[i])
```
Comment (CaedenPH) [inline machine_learning/linear_regression.py:98]:
```suggestion
    total = sum(abs(original_y[i] - predicted_y[i]) for i in range(len(original_y)))
```

**Review by CaedenPH — COMMENTED** [review]
@cclauss Looks good - perhaps you commit my changes

File: machine_learning/linear_regression.py [hunk 4]
```diff
@@ -87,6 +87,16 @@ def run_linear_regression(data_x, data_y):
     return theta
 
 
+def mean_absolute_error(predicted_y, original_y):
+    """Return sum of square error for error calculation
+    :param predicted_y   : contains the output of prediction (result vector)
+    :param original_y    : contains values of expected outcome
+    :return          : mean absolute error computed from given feature's
+    """
+    total = sum(abs(original_y[i] - predicted_y[i]) for i in range(len(original_y)))
```
Comment (cclauss) [inline machine_learning/linear_regression.py:96]:
```suggestion
    total = sum(abs(y - predicted_y[i]) for i, y in enumerate(original_y))
```

**Review by cclauss — APPROVED** [review]

**Pull request merged**